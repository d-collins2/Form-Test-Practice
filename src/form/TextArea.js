import React from 'react';

const TextArea = ({ value, handleChange, formError, name, controlId }) => {
    function errorClass(error) {
        if(error){
            return( error.length === 0 ? '' : 'has-error');
        }
    }
    return (
        <div className={`form-group ${errorClass(formError)}`} id={controlId}>
            <label>{controlId}</label>
            <textarea
                required
                className="form-control"
                onChange={handleChange}
                rows="4"
                name={name}
                value={value}/>
            <span className="help-block text-red">{formError}</span>
            <div className="text-right">
                <span><small className="text-muted">Required</small></span>
            </div>
        </div>
    )
}

export default TextArea;
