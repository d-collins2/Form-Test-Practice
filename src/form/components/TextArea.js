import React from 'react';
import { Form } from 'react-bootstrap'

const TextArea = ({ controlId, formError, handleChange, name, value }) => {
    function errorClass(error) {
        if(error){
            return( error.length === 0 ? '' : 'has-error');
        }
    }
    return (
        <Form.Group className={errorClass(formError)} id={controlId}>
            <Form.Label>{controlId}</Form.Label>
            <Form.Control
                required
                as="textarea"
                name={name}
                onChange={handleChange}
                rows="4"
                value={value}/>
            <Form.Text className="help-block text-red">{formError}</Form.Text>
            <div className="text-right">
                <Form.Text className="text-muted">Required</Form.Text>
            </div>
        </Form.Group>
    )
}

export default TextArea;
