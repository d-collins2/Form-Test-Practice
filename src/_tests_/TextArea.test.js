import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'react-bootstrap'
import { shallow, mount } from 'enzyme';
import TextArea from '../form/components/TextArea';

describe('<TextArea />', function () {
    it('Should render correctly', function () {
        const component = renderer.create(<TextArea />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should have the correct Text Area label', function(){
        const label = "label";
        const component = mount(<TextArea label={label}/>);
        expect(component.find(Form.Label).text()).toEqual(label);
    });

    it('Should call changeHandler onChange on input', function() {
        const mockHandler = jest.fn();
        const event = { target: { value: "sometext" } };
        const component = mount(<TextArea handleChange={mockHandler}/>);

        expect(component.find(Form.Control).simulate("change", event));
        expect(mockHandler.mock.calls.length).toBe(1);
    })
})
