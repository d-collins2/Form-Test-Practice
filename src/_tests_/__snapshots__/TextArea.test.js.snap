// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> Should render correctly 1`] = `
<div
  className="has-error form-group"
  id="Id"
>
  <label
    className="form-label"
  >
    Label
  </label>
  <textarea
    className="form-control"
    name="Name"
    onChange={[Function]}
    required={true}
    rows="4"
    value="Value"
  />
  <small
    className="help-block text-red form-text"
  >
    Error
  </small>
  <div
    className="text-right"
  >
    <small
      className="text-muted form-text"
    >
      Required
    </small>
  </div>
</div>
`;
